dao/src/main/java/org/thingsboard/server/dao/sql/JpaPartitionedAbstractDao.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright © 2016-2023 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.sql;
+
+import org.thingsboard.server.dao.model.BaseEntity;
+import org.thingsboard.server.dao.util.SqlDao;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+@SqlDao
+public abstract class JpaPartitionedAbstractDao<E extends BaseEntity<D>, D> extends JpaAbstractDao<E, D> {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
```

Comment:
Please add a detailed Java doc to provide a new person with a shortcut to get the idea ASAP.
Something like that
```
/**
* The initial implementation is taken from repository.save(). The only difference is that id is generated by us, and we provide the flag isNew for new entities instead of null id.
**/
```